src/datasources/cache/__tests__/fake.cache.service.ts

```diff
@@ -49,7 +51,25 @@ export class FakeCacheService implements ICacheService {
     if (fields === undefined) {
       this.cache[cacheDir.key] = {};
     }
-    this.cache[cacheDir.key][cacheDir.field] = value;
+    (this.cache[cacheDir.key] as Record<string, string>)[cacheDir.field] =
+      value;
     return Promise.resolve();
   }
+
+  increment(
+    cacheKey: string,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    expireTimeSeconds: number | undefined,
+  ): Promise<number> {
+    if (!this.cache[cacheKey]) {
+      this.cache[cacheKey] = 1;
+    } else {
+      this.cache[cacheKey] = ++(this.cache[cacheKey] as number);
+    }
+    return Promise.resolve(this.cache[cacheKey] as number);
+  }
+
+  initNumericalValue(cacheKey: string, value: number): void {
+    this.cache[cacheKey] = value;
+  }
```

Comment:
I find this function a little bit out of place as this is mostly setting a value. By extending the Fake with specialised functions we make it harder for the caller to know how they should use it.

For example if you replace the real cache with a Fake one, the callers will only call `increment` because `initNumericalValue` is not part of interface being offered. 

Shouldn't this behaviour be covered already by the `increment` implementation?

Reply:
I agree with you, but this is needed in order to test the case where there is a pre-existent value behind a key, and it gets incremented, as @iamacook pointed out in [this comment](https://github.com/safe-global/safe-client-gateway/pull/1225#discussion_r1507292230): 

> add a second test that has a pre-existing value in the cache, starting from a higher number.

With the current set of functions exposed by `FakeCacheService`, we don't have a way to set a numeric value behind a key. The interface only provides a way to `set` by passing a `CacheDir`, i.e.: setting a Redis hash, not a simple key-value. 

This additional function is not needed in the `RedisCacheService` as we have the actual Redis client available in the tests, so we can _bypass_ the service.

I can see that this approach is not super consistent, as it introduces differences between the implementations (even considering `FakeCacheService` is test-only). I'm open to alternatives to solve this, some that comes to my mind are:
- Expose the `cache` class variable from `FakeCacheService` to the outside world, to be able to modify its state from outside the class.
- Add a `set` function to both implementations (and remove this `initNumericValue` then) that maps Redis `SET` command. The already existent `set` functions are mapping `HSET` right now, so this would be a wider refactor.
- Do not test the case that @iamacook requested (see above).

Do you have ny preference/additional solution @fmrsabino ?

Reply:
OK, I've added your suggestion. I agree it's not perfect because based on the type of the value provided, the cache will be written differently, so, for example, writing a `string` in the cache's _first level_ is not possible, and also to write a `number` into the _second level_ (inside the _Redis hash_ this class mimics). But I agree it's sufficient and maybe a more consistent one.

I guess the ideal solution if we want to write both plain key-values and hashes (something that we didn't need before now), would be to map Redis `set` and `hset` 🤔

Reply:
Yes, I believe that for now this solution works but we need to come back to it to provide a more robust implementation of the fake. No need to tackle in this PR 👍

Reply:
Changed in [797c9f2](https://github.com/safe-global/safe-client-gateway/pull/1225/commits/797c9f211d41722511d547963477ff597e208f8e)

Thanks!